### PR TITLE
[4.4] Feed: Fixing link selection of AtomParser

### PR DESCRIPTION
**Pull request by Hackwar**
Pull Request for Issue #38631 .

### Summary of Changes
The feed parser for Atom feeds is selecting the wrong link as described in the original issue.

### Testing Instructions
Copied from the original issue:
 1. Create new Feed Display module
 2. Set Feed URL to a Blogger Atom Feed eg `https://www.failureasaservice.com/feeds/posts/default` 

### Actual result BEFORE applying this Pull Request
Link is created using the first entry `<link rel="replies">` which is an XML feed and not useful for normal users who just want to jump over to your post on Blogger.com.

```
<link rel="replies" type="application/atom+xml" href="https://www.failureasaservice.com/feeds/921417709640098202/comments/default" title="Post Comments"/>
    <link rel="replies" type="text/html" href="https://www.failureasaservice.com/2020/10/amd-killed-arm-in-data-center.html#comment-form" title="0 Comments"/>
    <link rel="edit" type="application/atom+xml" href="https://www.blogger.com/feeds/3991757369876144735/posts/default/921417709640098202"/>
    <link rel="self" type="application/atom+xml" href="https://www.blogger.com/feeds/3991757369876144735/posts/default/921417709640098202"/>
    <link rel="alternate" type="text/html" href="https://www.failureasaservice.com/2020/10/amd-killed-arm-in-data-center.html" title="AMD Killed ARM in the Data Center"/>
```

### Expected result AFTER applying this Pull Request
Links should use the `<link rel="alternate">` entry creating a clickable jump back to the Blogger page.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by fgsw** [comment]
image

Without the pr a click on 'FaaS - Failure as a Service' and 'AMD Killed ARM in the Data Center' opens

image


PR applied only the first link opens the above window.

**Comment by TLWebdesign** [comment]
Hi @Hackwar i'm testing this PR but i don't see a change in the dom of the urls of the blog items in the feed. they all have rel="noopener" even after applying the patch. But i do notice a change in behaviour as that now it opens the actual article instead of the feed link when clicking on the items. Seems like a half successfull test no?

**Comment by TLWebdesign** [comment]
I have tested this item :white_check_mark: successfully on 231e36014e46e47faf3460ed9602078e1666083b<br /><br />Tested successfully <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42706">issues.joomla.org/tracker/joomla-cms/42706</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42706">issues.joomla.org/tracker/joomla-cms/42706</a>.</sub>

File: libraries/src/Feed/Parser/AtomParser.php [hunk 1]
```diff
@@ -238,13 +238,13 @@ protected function processFeedEntry(FeedEntry $entry, \SimpleXMLElement $el)
     /**
      * If there is more than one <link> in the feed entry, find the most appropriate one and return it.
      *
-     * @param   array  $links  Array of <link> elements from the feed entry.
+     * @param   \SimpleXMLElement  $links  Array of <link> elements from the feed entry.
```
Comment (laoneo) [inline libraries/src/Feed/Parser/AtomParser.php:241]: Can you change the description here as it is not an array anymore.

Comment (laoneo) [inline libraries/src/Feed/Parser/AtomParser.php]:
```suggestion
     * @param   \SimpleXMLElement  $links  XML node with links from the feed entry.
```

**Comment by MacJoom** [comment]
Thank you!

**Pull request merged**